packages/type-safe-api/src/construct/spec/utils.ts

```diff
@@ -16,3 +17,19 @@ export const functionInvocationUri = (
   const stack = Stack.of(scope);
   return `arn:${stack.partition}:apigateway:${stack.region}:lambda:path/2015-03-31/functions/${lambdaFunction.functionArn}/invocations`;
 };
+
+/**
+ * Generate the s3 bucket invocation uri for the given s3 within the given scope
+ * @param scope scope in which the s3 is deployed
+ * @param bucket the s3 bucket to be invoked
+ */
+export const bucketInvocationUri = (
+  scope: Construct,
+  bucket: IBucket,
+  pathOverride?: string
+): string => {
+  const stack = Stack.of(scope);
+  return `arn:${stack.partition}:apigateway:${stack.region}:s3:path/${
```

Comment:
I can't find the relevant documentation re cross-region s3 integrations - is it definitely the case that we want to use the API's region in this uri (from the `scope` variable) rather than the region for the bucket?

Reply:
My bad you're right, there is not really a documentation about it, but if you try to configure an API like so in the console and export the API with API gateway extension you'll see that's the bucket region and so i supposed the bucket partition too.

You should probably update the `functionInvocationUri` too because it's the lambda region and partition too. I know you generate handlers in same region but in case a user might want to use a custom handler

Reply:
Ah nice one! Thanks for looking into this - I'll fix the lambdas! 😄